R/plot-build.r

```diff
@@ -167,7 +167,12 @@ ggplot_gtable.ggplot_built <- function(data) {
   theme <- plot_theme(plot)
 
   geom_grobs <- by_layer(function(l, d) l$draw_geom(d, layout), plot$layers, data, "converting geom to grob")
-  layout$setup_panel_guides(plot$guides, plot$layers, plot$mapping)
+
+  guides <- plot$guides
+  if (!inherits(guides, "Guides")) {
+    guides <- guides_list(guides)
```

Comment:
Is this ever possible except for old plot objects?

Reply:
Yeah `ggplot()` doesn't initialise any guide structure, so `plot$guides` may be `NULL` if you don't manually add guides.

Reply:
I think we should align it with how scales are handled and always initialise a `Guides` object, unless there is strong reasons not to